New version: DuckStudio.GitHubView.Nightly version 2025.01.22.12907017964

Dragon1573 opened the pull request:
## Checklist for Pull Requests

- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Resolve #215065

## Manifests

- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/doc/Authoring.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.9 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.9.0)?

Note: `<path>` is the directory's name containing the manifest you're submitting.

## Manual validation

```text
--> Installing WinGet
--> Disabling safety warning when running installers
Tip: you can type 'Update-EnvironmentVariables' to update your environment variables, such as after installing a new software.

--> Configuring Winget
Enabled admin setting 'LocalManifestFiles'.
Enabled admin setting 'LocalArchiveMalwareScanOverride'.

--> Installing the Manifest 2025.01.22.12907017964

Found GitHubView [DuckStudio.GitHubView.Nightly] Version 2025.01.22.12907017964
This application is licensed to you by its owner.
Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
This package requires the following dependencies:
  - Packages
      Microsoft.DotNet.Runtime.8
(1/1) Found Microsoft .NET Runtime 8.0 [Microsoft.DotNet.Runtime.8] Version 8.0.12
This application is licensed to you by its owner.
Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
Downloading https://builds.dotnet.microsoft.com/dotnet/Runtime/8.0.12/dotnet-runtime-8.0.12-win-x64.exe
  ██████████████████████████████  27.2 MB / 27.2 MB
Successfully verified installer hash
Starting package install...
Successfully installed

Downloading https://github.com/DuckDuckStudio/GitHubView/releases/download/2025.01.22.12907017964/GitHubView-2025.01.22.12907017964-windows-latest-x64-Setup.exe
  ██████████████████████████████  2.04 MB / 2.04 MB
Successfully verified installer hash
Starting package install...
Successfully installed

--> Refreshing environment variables

--> Comparing ARP Entries

DisplayName                           DisplayVersion         Publisher             ProductCode                            Scope
-----------                           --------------         ---------             -----------                            -----
GitHubView 版本 2025.01.22.12907017964  2025.01.22.12907017964 鸭鸭「カモ」                GitHubView_is1                         Machine
Microsoft .NET Runtime - 8.0.12 (x64) 8.0.12.34403           Microsoft Corporation {ba2e9943-e30e-49df-89e8-7db7905c6cf0} Machine
```

-----

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/215274)

Dragon1573 committed:
New version: DuckStudio.GitHubView.Nightly version 2025.01.22.12907017964

Signed-off-by: Dragon1573 <49941141+Dragon1573@users.noreply.github.com>

Dragon1573 commented:
> [!IMPORTANT]
> 
> Should be reviewed by @DuckDuckStudio 🙏🏼

DuckDuckStudio commented on manifests/d/DuckStudio/GitHubView/Nightly/2025.01.22.12907017964/DuckStudio.GitHubView.Nightly.installer.yaml:
```diff
@@ -0,0 +1,39 @@
+
+PackageIdentifier: DuckStudio.GitHubView.Nightly
+PackageVersion: 2025.01.22.12907017964
+InstallerLocale: en-US
```
```suggestion
InstallerLocale: zh-CN
```

![image](https://github.com/user-attachments/assets/aa4214eb-6c47-4fea-8cd2-7ba8b596dfb2)
我不确定应该填哪个语言，但之前的清单中我填 zh-CN
![image](https://github.com/user-attachments/assets/7f358ba6-d1f5-4ea1-b813-3d7308d97fc4)
(在 iss 文件中)

DuckDuckStudio replied:
英文语言的`is1`叫Default，但我没有明确指定哪种语言为默认。

DuckDuckStudio commented on manifests/d/DuckStudio/GitHubView/Nightly/2025.01.22.12907017964/DuckStudio.GitHubView.Nightly.installer.yaml:
```diff
@@ -0,0 +1,39 @@
+Dependencies:
+  PackageDependencies:
+  - PackageIdentifier: Microsoft.DotNet.Runtime.8
+ProductCode: GitHubView_is1
+ReleaseDate: 2025-01-22
+AppsAndFeaturesEntries:
+- ProductCode: GitHubView_is1
+ElevationRequirement: elevatesSelf
```
```suggestion
ReleaseDate: 2025-01-22
```

我在 iss 文件中没有指定 ProductCode。
当程序以用户范围安装时，不需要提升权限。

DuckDuckStudio replied:
抱歉，我没有发现这个安装程序和我之前几个程序的安装程序的“独特之处”
- https://github.com/microsoft/winget-pkgs/pull/215274#discussion_r1935976599

DuckDuckStudio commented on manifests/d/DuckStudio/GitHubView/Nightly/2025.01.22.12907017964/DuckStudio.GitHubView.Nightly.locale.zh-CN.yaml:
```diff
@@ -0,0 +1,41 @@
+Author: 鸭鸭「カモ」
+PackageName: GitHubView
+PackageUrl: https://github.com/DuckDuckStudio/GitHubView/
+License: GPL-3.0
+LicenseUrl: https://github.com/DuckDuckStudio/GitHubView/blob/HEAD/LICENSE.txt
```
```suggestion
License: GNU General Public License v3.0
LicenseUrl: https://github.com/DuckDuckStudio/GitHubView/blob/master/LICENSE.txt
```

与英文清单保持一致

DuckDuckStudio commented on manifests/d/DuckStudio/GitHubView/Nightly/2025.01.22.12907017964/DuckStudio.GitHubView.Nightly.locale.en-US.yaml:
```diff
@@ -0,0 +1,44 @@
+- Nightly
+- Terminal-tool
+- Tool
+ReleaseNotes: |-
+  ## What's Changed
+    - Updated Development Documentation by @DuckDuckStudio in #12
+
+  **Full Changelog**: https://github.com/DuckDuckStudio/GitHubView/compare/2025.01.14.12764289055...2025.01.15.12782147412
```
```suggestion
ReleaseNotes: |-
  What's Changed
  - refactor(build): Refactoring the build release workflow by @DuckDuckStudio in #18
  Full Changelog: https://github.com/DuckDuckStudio/GitHubView/compare/2025.01.19.12849721517...2025.01.22.12907017964
ReleaseNotesUrl: https://github.com/DuckDuckStudio/GitHubView/releases/tag/2025.01.22.12907017964
```

错误的发行说明

Dragon1573 committed:
Apply suggestions from code review

- Remove "ProductCode" from the manifest, since it's not manually configured

Co-authored-by: 鸭鸭「カモ」 <Yzcbs123@163.com>

Dragon1573 committed:
Apply suggestions from code review

- Update "Release Notes"
- Update "License" and its URL, to keep syncing with the en-US version

Co-authored-by: 鸭鸭「カモ」 <Yzcbs123@163.com>

Dragon1573 committed:
Apply suggestions from code review

- Switch the "Installer Locale" according to suggestions.

Co-authored-by: 鸭鸭「カモ」 <Yzcbs123@163.com>

DuckDuckStudio commented on manifests/d/DuckStudio/GitHubView/Nightly/2025.01.22.12907017964/DuckStudio.GitHubView.Nightly.installer.yaml:
```diff
@@ -0,0 +1,35 @@
+PackageVersion: 2025.01.22.12907017964
+InstallerLocale: zh-CN
+InstallerType: inno
+Scope: machine
```
```suggestion
```

stephengillie replied:
Should the `Scope` be modified?

DuckDuckStudio replied:
This is a leftover from a comment https://github.com/microsoft/winget-pkgs/pull/215274#discussion_r1925447996 I didn't change request because I didn't think it mattered whether I changed it or not.

stephengillie replied:
> This is a leftover from a comment [#215274 (comment)](https://github.com/microsoft/winget-pkgs/pull/215274#discussion_r1925447996) 

Ah, thanks. 

> edit: 如果设置了机器范围，用户该如何将其安装到用户范围？

(Internet Translation: If it is set machine-wide, how can a user install it to user-scope?)

The user can't. To perform this, the set of `Installers` should be duplicated. One set should have `Scope: machine` specified, and `Scope: user` for the other.

> > 当程序以用户范围安装时，不需要提升权限

(Internet Translation: When the program is installed as a user, elevated privileges are not required)

This is good to know. The `Installers` set described above, with the entry `Scope: machine` should also have `ElevationRequirement: elevatesSelf`.

DuckDuckStudio replied:
> To perform this, the set of should be duplicated. One set should have specified, and for the other.`Installers``Scope: machine``Scope: user`

In my previous manifest, I simply removed the scope entry.

stephengillie replied:
I forgot to mention this part - installer switches (or another means) would be needed to specify which scope. If the installer only installs to machine scope, then that value doesn't matter.

DuckDuckStudio replied:
( [Conversation needs to be resolved by PR author or maintainer markup](https://github.com/orgs/community/discussions/148165) )

DuckDuckStudio commented:
The reason for the label https://github.com/microsoft/winget-pkgs/labels/Validation-Installation-Error 's appearance remains a mystery...

stephengillie commented:
Automatic Validation ended with:
>  Installation failed with exit code -1978335187 2025-01-27 20:25:35.965 [FAIL] Installer failed security check. Url: https://github.com/DuckDuckStudio/GitHubView/releases/download/2025.01.22.12907017964/GitHubView-2025.01.22.12907017964-windows-latest-x64-Setup.exe Result: 0x80004005 MSI (s) (A4:18) [20:25:16:332]: PROPERTY CHANGE: Adding OutOfDiskSpace property. Its value is '0'. MSI (s) (A4:18) [20:25:16:332]: PROPERTY CHANGE: Adding OutOfNoRbDiskSpace property. Its value is '0'. MSI (s) (A4:18) [20:25:16:332]: PROPERTY CHANGE: Adding PrimaryVolumeSpaceAvailable property. Its value is '0'. MSI (s) (A4:18) [20:25:16:332]: PROPERTY CHANGE: Adding PrimaryVolumeSpaceRequired property. Its value is '0'. MSI (s) (A4:18) [20:25:16:332]: PROPERTY CHANGE: Adding PrimaryVolumeSpaceRemaining property. Its value is '0'. MSI (s) (A4:18) [20:25:17:362]: Executing op: ComponentRegister(ComponentId={4C4F177D-EAA8-526E-B5BA-A435AB8AE909},KeyPath=22*85*\SOFTWARE\Microsoft\Windows\Windows Error Reporting\RuntimeExceptionHelperModules\C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.12\mscordaccore.dll,State=3,,Disk=1,SharedDllRefCount=0,BinaryType=1) MSI (s) (A4:18) [20:25:18:554]: Executing op: RegOpenKey(Root=-2147483646,Key=SOFTWARE\Microsoft\Windows\Windows Error Reporting\RuntimeExceptionHelperModules,,BinaryType=1,,) Property(S): ErrorDialog = ErrorDlg Property(S): OutOfDiskSpace = 0 Property(S): OutOfNoRbDiskSpace = 0 Property(S): PrimaryVolumeSpaceAvailable = 0 Property(S): PrimaryVolumeSpaceRequired = 0 Property(S): PrimaryVolumeSpaceRemaining = 0 MSI (s) (A4:6C) [20:25:23:017]: PROPERTY CHANGE: Adding OutOfDiskSpace property. Its value is '0'. MSI (s) (A4:6C) [20:25:23:017]: PROPERTY CHANGE: Adding OutOfNoRbDiskSpace property. Its value is '0'. MSI (s) (A4:6C) [20:25:23:017]: PROPERTY CHANGE: Adding PrimaryVolumeSpaceAvailable property. Its value is '0'. MSI (s) (A4:6C) [20:25:23:017]: PROPERTY CHANGE: Adding PrimaryVolumeSpaceRequired property. Its value is '0'. MSI (s) (A4:6C) [20:25:23:017]: PROPERTY CHANGE: Adding PrimaryVolumeSpaceRemaining property. Its value is '0'. MSI (s) (A4:A0) [20:25:27:204]: PROPERTY CHANGE: Adding OutOfDiskSpace property. Its value is '0'. MSI (s) (A4:A0) [20:25:27:204]: PROPERTY CHANGE: Adding OutOfNoRbDiskSpace property. Its value is '0'. MSI (s) (A4:A0) [20:25:27:204]: PROPERTY CHANGE: Adding PrimaryVolumeSpaceAvailable property. Its value is '0'. MSI (s) (A4:A0) [20:25:27:204]: PROPERTY CHANGE: Adding PrimaryVolumeSpaceRequired property. Its value is '0'. MSI (s) (A4:A0) [20:25:27:204]: PROPERTY CHANGE: Adding PrimaryVolumeSpaceRemaining property. Its value is '0'. 2025-01-27 20:17:27.432 [FAIL] Installer failed security check. Url: https://github.com/DuckDuckStudio/GitHubView/releases/download/2025.01.22.12907017964/GitHubView-2025.01.22.12907017964-windows-latest-x86-Setup.exe Result: 0x80004005 MSI (s) (44:F4) [20:17:19:688]: PROPERTY CHANGE: Adding OutOfDiskSpace property. Its value is '0'. MSI (s) (44:F4) [20:17:19:688]: PROPERTY CHANGE: Adding OutOfNoRbDiskSpace property. Its value is '0'. MSI (s) (44:F4) [20:17:19:688]: PROPERTY CHANGE: Adding PrimaryVolumeSpaceAvailable property. Its value is '0'. MSI (s) (44:F4) [20:17:19:688]: PROPERTY CHANGE: Adding PrimaryVolumeSpaceRequired property. Its value is '0'. MSI (s) (44:F4) [20:17:19:688]: PROPERTY CHANGE: Adding PrimaryVolumeSpaceRemaining property. Its value is '0'. MSI (s) (44:F4) [20:17:20:657]: Executing op: ComponentRegister(ComponentId={BE010199-713F-57C2-9E29-CAA2CA6BC85C},KeyPath=02*85*\SOFTWARE\Microsoft\Windows\Windows Error Reporting\RuntimeExceptionHelperModules\C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.12\mscordaccore.dll,State=3,,Disk=1,SharedDllRefCount=0,BinaryType=0) MSI (s) (44:F4) [20:17:21:860]: Executing op: RegOpenKey(Root=-2147483646,Key=SOFTWARE\Microsoft\Windows\Windows Error Reporting\RuntimeExceptionHelperModules,,BinaryType=0,,) MSI (s) (44:14) [20:17:22:220]: PROPERTY CHANGE: Adding OutOfDiskSpace property. Its value is '0'. MSI (s) (44:14) [20:17:22:220]: PROPERTY CHANGE: Adding OutOfNoRbDiskSpace property. Its value is '0'. MSI (s) (44:14) [20:17:22:220]: PROPERTY CHANGE: Adding PrimaryVolumeSpaceAvailable property. Its value is '0'. MSI (s) (44:14) [20:17:22:220]: PROPERTY CHANGE: Adding PrimaryVolumeSpaceRequired property. Its value is '0'. MSI (s) (44:14) [20:17:22:220]: PROPERTY CHANGE: Adding PrimaryVolumeSpaceRemaining property. Its value is '0'. MSI (s) (44:A0) [20:17:22:688]: PROPERTY CHANGE: Adding OutOfDiskSpace property. Its value is '0'. MSI (s) (44:A0) [20:17:22:688]: PROPERTY CHANGE: Adding OutOfNoRbDiskSpace property. Its value is '0'. MSI (s) (44:A0) [20:17:22:688]: PROPERTY CHANGE: Adding PrimaryVolumeSpaceAvailable property. Its value is '0'. MSI (s) (44:A0) [20:17:22:688]: PROPERTY CHANGE: Adding PrimaryVolumeSpaceRequired property. Its value is '0'. MSI (s) (44:A0) [20:17:22:688]: PROPERTY CHANGE: Adding PrimaryVolumeSpaceRemaining property. Its value is '0'.

Automated error analysis suggests -1978335187 may mean Installer failed security check.  <!-- [Policy] Validation-Defender-Error -->

Automated error analysis suggests 80004005 may mean Resource missing

(Automated response - build 959.)

DuckDuckStudio commented:
Hi @Dragon1573 ,
Can you reopen this PR? Usually the issue is resolved after re-running the pipeline a few times.
- Related to https://github.com/microsoft/winget-pkgs/pull/217501#issuecomment-2619722573

Dragon1573 commented:
> Can you reopen this PR? Usually the issue is resolved after re-running the pipeline a few times.

Okay ... I'll try if I can re-open it.

DuckDuckStudio commented:
> - Related to https://github.com/microsoft/winget-pkgs/pull/217501#issuecomment-2619722573
> ### ✏️ Edit
> Emmm ... It seems I have no authority to re-open the pull request. I can only restore the deleted branch.
> 
> This pull request is closed by the bot [**@microsoft-github-policy-service**](https://github.com/apps/microsoft-github-policy-service) instead of me. We can only ask if @stephengillie (or other [moderators](https://github.com/microsoft/winget-pkgs/issues/15674)) could help. 😞

![image](https://github.com/user-attachments/assets/4a73621d-a64d-4bb3-b847-a3349d184d1b)

DuckDuckStudio commented:
> === manifests\d\DuckStudio\GitHubView\Nightly\2025.01.22.12907017964 ===  
> Missing Properties value based on version 2025.01.19.12849721517:  
> Installers

This version does not have an installer for the arm architecture.

stephengillie commented:
https://github.com/microsoft/winget-pkgs/pull/215274#discussion_r1934685964
